### PR TITLE
send message to channel when joined run has private channel

### PR DESCRIPTION
**Pull request by trilopin**
#### Summary
When some user joins the run through participate but the channel is private, user is added to the run but not to the channel. Same happens when the user who add another new user to the run doesn't have access to the private channel.

For those cases, a message will be sent to the channel as playbooks bot.

cleanup:
- `leave` and `requestGetInvolved` REST run endpoints removed, also removed from client.ts
- `requestGetInvolved` service removed, part of it done in `participateActions`
- `leave` and `requestGetInvolved` removed from client

**Graphql & client**
I was tempted to add the functions addParticipants/removeParticipants to the client package, but I ended up thinking that graphql/schema is already the API (and we should not restrict/hide to API clients). I've just added a shortcut to handle the mutations more easily in our server tests (and not write them each time).

0/5 on the messages, they need more work

#### Ticket Link
Fixes: https://github.com/mattermost/mattermost-plugin-playbooks/issues/1434

#### Checklist
- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [x] Unit tests updated

### CODE REVIEW

File: assets/i18n/en.json [hunk 1]
```diff
@@ -86,8 +86,12 @@
...
   },
   {
-    "id": "app.user.run.request_get_involved",
-    "translation": "@here — @{{.Name}} wants to participate in this run. To include them as a participant, add them to this channel.\n"
+    "id": "app.user.run.joined_run_channel_private_add_participant",
+    "translation": "@here — @{{.Name}} has been added to the run by @{{.RequesterName}}. They haven't been automatically added to the channel because @{{.RequesterName}} is not a channel member and the channel is private.\n"
+  },
+  {
+    "id": "app.user.run.joined_run_channel_private_participate",
+    "translation": "@here — @{{.Name}} has joined the run. They haven't been automatically added to the channel because it's private.\n"
```
Comment (trilopin) [inline assets/i18n/en.json:94]: context: https://community.mattermost.com/core/pl/o3zhgbs1b3rmf89grhjqzf7zxa

suggestions welcome, these messages are not ready @itao @justinegeffen

**Review by gbochora — APPROVED** [review]
Looks good! Thanks @trilopin

File: assets/i18n/en.json [hunk 2]
```diff
@@ -86,8 +86,12 @@
     "translation": "You have 0 assigned tasks."
   },
   {
-    "id": "app.user.run.request_get_involved",
-    "translation": "@here — @{{.Name}} wants to participate in this run. To include them as a participant, add them to this channel.\n"
+    "id": "app.user.run.joined_run_channel_private_add_participant",
+    "translation": "@here — @{{.Name}} has been added to the run by @{{.RequesterName}}. They haven't been automatically added to the channel because @{{.RequesterName}} is not a channel member and the channel is private.\n"
```
Comment (crspeller) [inline assets/i18n/en.json:90]: This feels like it's a bit noisy. Are we sure we want `@here`?

**Review by crspeller — APPROVED** [review]

File: assets/i18n/en.json (hunk 2 above)
Comment (trilopin) [reply assets/i18n/en.json:90]: there was already here, but 1/5 on that one

**Comment by justinegeffen** [comment]
Sorry I missed this, @trilopin! I was out of the office. Let me know if you still need my review. :)

**Comment by trilopin** [comment]
> Sorry I missed this, @trilopin! I was out of the office. Let me know if you still need my review. :)

no problem @justinegeffen, your help would be very appreciated on https://github.com/mattermost/mattermost-plugin-playbooks/pull/1440

**Pull request merged**